src/plugins/mer/mersdkmanager.cpp

```diff
@@ -580,17 +580,24 @@ void MerSdkManager::ensureCmakeToolIsSet(Kit *k, const BuildEngine *buildEngine,
     const FileName cmakeWrapper = FileName(buildTarget.toolsPath).appendPath(Sfdk::Constants::WRAPPER_CMAKE);
 
     if (cmakeWrapper.exists()) {
-        const Core::Id cmakeId = CMakeToolManager::registerOrFindCMakeTool(cmakeWrapper);
-        CMakeTool *cmakeTool = CMakeToolManager::findById(cmakeId);
-        cmakeTool->setDisplayName(QString::fromLatin1("CMake for %1 in %2").arg(buildTargetName, buildEngine->name()));
-        cmakeTool->setAutorun(true);
-        cmakeTool->setAutoCreateBuildDirectory(true);
-        CMakeKitInformation::setCMakeTool(k, cmakeId);
-        QStringList cmakeConf = { "CMAKE_CXX_COMPILER:STRING=/usr/bin/gcc",
-                                  "CMAKE_C_COMPILER:STRING=/usr/bin/gcc",
-                                  "CMAKE_PREFIX_PATH:STRING=%{Qt:QT_INSTALL_PREFIX}",
-                                  "QT_QMAKE_EXECUTABLE:STRING=%{Qt:qmakeExecutable}" };
-        CMakeConfigurationKitInformation::fromStringList(k, cmakeConf);
+        if (const CMakeTool *existing = CMakeToolManager::findByCommand(cmakeWrapper)) {
+            CMakeKitInformation::setCMakeTool(k, existing->id());
+        } else {
+            auto cmakeTool = std::make_unique<CMakeTool>(CMakeTool::AutoDetectionByPlugin, CMakeTool::createId());
+            cmakeTool->setCMakeExecutable(cmakeWrapper);
+            cmakeTool->setDisplayName(QString::fromLatin1("CMake for %1 in %2").arg(buildTargetName, buildEngine->name()));
+            cmakeTool->setAutorun(true);
+            cmakeTool->setAutoCreateBuildDirectory(true);
+            const Core::Id id = cmakeTool->id();
+            const bool registeredOk = CMakeToolManager::registerCMakeTool(std::move(cmakeTool));
+            QTC_ASSERT(registeredOk, return);
+            CMakeKitInformation::setCMakeTool(k, id);
+            QStringList cmakeConf = { "CMAKE_CXX_COMPILER:STRING=/usr/bin/gcc",
+                                      "CMAKE_C_COMPILER:STRING=/usr/bin/gcc",
+                                      "CMAKE_PREFIX_PATH:STRING=%{Qt:QT_INSTALL_PREFIX}",
+                                      "QT_QMAKE_EXECUTABLE:STRING=%{Qt:qmakeExecutable}" };
```

Comment:
I have a feeling that these should point to the wrapper scripts

Reply:
To me it looks suspicious too but I couldn't fully understand the effect quickly. Definitely something to keep in mind and possibly verify, but out of scope here.